### PR TITLE
Update PT to PL conversion doc

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/starter/converting.rst [hunk 1]
```diff
@@ -76,38 +85,121 @@ To add an (optional) validation loop add logic to the
...
+            pred = self(x)
+            return pred
+
+.. note:: ``model.eval()`` and ``torch.no_grad()`` are called automatically for prediction.
+
+The predict loop will not be used until you call.
+
+.. code-block:: python
+
+    trainer.predict()
```
Comment (Borda) [inline docs/source/starter/converting.rst:144]: it does not need any argument?

File: docs/source/starter/converting.rst [hunk 2]
```diff
@@ -76,38 +85,121 @@ To add an (optional) validation loop add logic to the
...
-Your :doc:`lightning module <../common/lightning_module>` can automatically run on any hardware!
+**************
+8. Plugin Data
+**************
+
+To plugin your DataLoaders, you can override the respective dataloader hooks:
+
+.. testcode::
+
+    class LitModel(LightningModule):
```
Comment (Borda) [inline docs/source/starter/converting.rst:187]: I thought this is not recommended way...

Comment (rohitgr7) [reply docs/source/starter/converting.rst:187]: what do you mean?

File: docs/source/starter/converting.rst [hunk 3]
```diff
@@ -64,10 +71,12 @@ This should go into the :func:`~pytorch_lightning.core.LightningModule.training_
...
 
-4. Find the val loop "meat"
-===========================
+***************************
+4. Find the Val Loop "meat"
+***************************
+
 To add an (optional) validation loop add logic to the
-:func:`~pytorch_lightning.core.LightningModule.validation_step` hook (make sure to use the hook parameters, ``batch`` and ``batch_idx`` in this case).
+:meth:`~pytorch_lightning.core.lightning.LightningModule.validation_step` hook (make sure to use the hook parameters, ``batch`` and ``batch_idx`` in this case).
```
Comment (tchaton) [inline docs/source/starter/converting.rst:80]:
```suggestion
:meth:`~pytorch_lightning.core.lightning.LightningModule.validation_step` hook (make sure to use the hook parameters, ``batch`` and ``batch_idx`` in this case).

```

File: docs/source/starter/converting.rst [hunk 4]
```diff
@@ -48,10 +53,12 @@ Move your optimizers to the :func:`~pytorch_lightning.core.LightningModule.confi
 
 --------
 
-3. Find the train loop "meat"
-=============================
-Lightning automates most of the training for you, the epoch and batch iterations, all you need to keep is the training step logic.
-This should go into the :func:`~pytorch_lightning.core.LightningModule.training_step` hook (make sure to use the hook parameters, ``batch`` and ``batch_idx`` in this case):
+*****************************
+3. Find the Train Loop "meat"
```
[the remark this answers is not in the document]
Comment (rohitgr7) [reply docs/source/starter/converting.rst:57]: `Configure Training Logic`?

File: docs/source/starter/converting.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/starter/converting.rst:80]: sorry, what's the suggestion here?

File: docs/source/starter/converting.rst [hunk 5]
```diff
@@ -76,38 +95,136 @@ To add an (optional) validation loop add logic to the
...
 --------
 
-6. Remove any .cuda() or to.device() calls
-==========================================
-Your :doc:`lightning module <../common/lightning_module>` can automatically run on any hardware!
+**************
+8. Plugin Data
+**************
+
+To plugin your DataLoaders, you can override the respective dataloader hooks:
```
Comment (tchaton) [inline docs/source/starter/converting.rst:208]:
```suggestion
**********************
8. Use your own data
**********************

To use your DataLoaders, you can override the respective dataloader hooks on the :class:`~pytorch_lightning.core.lightning.LightningModule`:
```

File: docs/source/starter/converting.rst [hunk 6]
```diff
@@ -76,38 +95,136 @@ To add an (optional) validation loop add logic to the
...
+
+.. tip:: ``trainer.test()`` loads the best checkpoint automatically by default if checkpointing is enabled.
+
+--------
+
+*****************************
+6. Configure Prediction Logic
+*****************************
+
+Lightning automates the prediction loop for you and manages all the associated components, such as epoch and batch tracking. As a user,
```
Comment (ethanwharris) [inline docs/source/starter/converting.rst:148]:
```suggestion
Lightning automates the prediction loop for you and manages all of the associated components such as epoch and batch tracking. As a user,
```

File: docs/source/starter/converting.rst [hunk 7]
```diff
@@ -76,38 +95,136 @@ To add an (optional) validation loop add logic to the
...
+        def val_dataloader(self):
+            return DataLoader(...)
+
+        def test_dataloader(self):
+            return DataLoader(...)
+
+        def predict_dataloader(self):
+            return DataLoader(...)
+
+Additionally, you can also plugin your dataloaders using one of the following ways:
```
Comment (ethanwharris) [inline docs/source/starter/converting.rst:225]:
```suggestion
Alternatively, you can pass your dataloaders in one of the following ways:
```

File: docs/source/starter/converting.rst [hunk 8]
```diff
@@ -76,38 +95,136 @@ To add an (optional) validation loop add logic to the
...
+
+* Pass in the dataloaders explictly inside ``trainer.fit/.validate/.test/.predict`` calls.
+* Use :ref:`LightningDataModule <datamodules>`.
```
Comment (ethanwharris) [inline docs/source/starter/converting.rst:228]:
```suggestion
* Use a :ref:`LightningDataModule <datamodules>`.
```

File: docs/source/starter/converting.rst [hunk 9]
```diff
@@ -64,10 +76,17 @@ This should go into the :func:`~pytorch_lightning.core.LightningModule.training_
 
 --------
 
-4. Find the val loop "meat"
-===========================
+*********************************
+4. Configure the Validation Logic
+*********************************
+
+Lightning also automates the validation loop for you and manages all the associated components, such as epoch and batch tracking, metrics reduction. As a user,
```
Comment (ethanwharris) [inline docs/source/starter/converting.rst:83]:
```suggestion
Lightning also automates the validation loop for you and manages all of the associated components such as: epoch and batch tracking, and metrics reduction. As a user,
```

**Review by ethanwharris — APPROVED** [review]
LGTM, some small suggestions
